tests/test_route53/test_route53.py

```diff
@@ -113,24 +117,35 @@ def test_rrset():
     changes.commit()
 
     changes = ResourceRecordSets(conn, zoneid)
+    change = changes.add_change("CREATE", "bar.foo.testdns.aws.com", "TXT")
+    change.add_value("bar")
     change = changes.add_change("CREATE", "foo.bar.testdns.aws.com", "A")
     change.add_value("1.2.3.4")
     change = changes.add_change("CREATE", "bar.foo.testdns.aws.com", "A")
     change.add_value("5.6.7.8")
     changes.commit()
 
-    rrsets = conn.get_all_rrsets(zoneid, type="A")
-    rrsets.should.have.length_of(2)
-
-    rrsets = conn.get_all_rrsets(zoneid, name="bar.foo.testdns.aws.com", type="A")
-    rrsets.should.have.length_of(1)
-    rrsets[0].resource_records[0].should.equal("5.6.7.8")
+    rrsets = conn.get_all_rrsets(zoneid)
+    rrsets.should.have.length_of(3)
+    rrsets[0].resource_records[0].should.equal("1.2.3.4")
+    rrsets[1].resource_records[0].should.equal("5.6.7.8")
+    rrsets[2].resource_records[0].should.equal("bar")
 
     rrsets = conn.get_all_rrsets(zoneid, name="foo.bar.testdns.aws.com", type="A")
-    rrsets.should.have.length_of(2)
+    rrsets.should.have.length_of(3)
     resource_records = [rr for rr_set in rrsets for rr in rr_set.resource_records]
     resource_records.should.contain("1.2.3.4")
     resource_records.should.contain("5.6.7.8")
+    resource_records.should.contain("bar")
```

Comment:
This should fail now, as type=A does not match the provided tye TXT.

Can we change the existing assertions here, to validate the order in which the items are returned?
As far as I can tell, there is no definitive validation yet that the new `sorted(..)` logic works, but this request would be a good test

Reply:
https://github.com/spulec/moto/pull/3864/files#diff-a426c01b3d9e83d976f9c3f7a209b1f988f37da5cfe37e1546100cd9606d9994R128-R132

This one tests that records have correct order. Will it be enough?

Reply:
According to docs types should not match. You need to return all the types that are greater or equal to the start_type.

Reply:
Yup - I misread the documentation. All good.